client/src/components/shared/types/index.ts

```diff
@@ -0,0 +1,7 @@
+export interface CalendarKeyDateEvent {
+  _id?: string;
```

Comment:
_id is provided by default in mongoDB, change this to create a custom field called id which is separate from _id in mongoDB